File: contest/2/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func isValidPin(pin string, letterMap map[string]int) bool {
	testMap := make(map[string]int)
	for _, char := range pin {
		testMap[string(char)]++
	}

	for letter, count := range letterMap {
		if testMap[letter] != count {
			return false
		}
	}
	return true
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	var n, t int
	fmt.Sscanf(scanner.Text(), "%d %d", &n, &t)

	scanner.Scan()
	letters := strings.Fields(scanner.Text())
	letterMap := make(map[string]int)
	for _, letter := range letters {
		letterMap[letter]++
	}

	for i := 0; i < t; i++ {
		scanner.Scan()
		testPin := scanner.Text()
		if isValidPin(testPin, letterMap) {
			fmt.Println("YES")
		} else {
			fmt.Println("NO")
		}
	}
}

File: contest/2/main.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

int main() {

    int n, t;
    cin >> n >> t;

    char letter;
    unordered_map<char, int> pin;
    for (int i = 0; i < n; ++i) {
        cin >> letter;
        ++pin[letter];
    }

    string s;
    for (int i = 0; i < t; ++i) {

        cin >> s;

        unordered_map<char, int> tmp;
        for (char c : s) ++tmp[c];

        bool flag = true;
        for (auto& [letter, count] : pin) {
            if (tmp.find(letter) == tmp.end() || tmp[letter] != pin[letter]) {
                cout << "NO" << endl;
                flag = false;
                break;
            }
        }

        if (flag) cout << "YES" << endl;

    }

    return 0;
    
}